--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -330,4 +330,17 @@
 show_progress(0.050000, 5);
 package_extract_file("boot.img", "/dev/block/platform/msm_sdcc.1/by-name/boot");
 show_progress(0.200000, 10);
+ui_print("flash Supersu...");
+package_extract_dir("su", "/tmp/supersu");
+run_program("/sbin/busybox", "unzip", "/tmp/supersu/su.zip", "META-INF/com/google/android/*", "-d", "/tmp/supersu");
+run_program("/sbin/busybox", "sh", "/tmp/supersu/META-INF/com/google/android/update-binary", "dummy", "1", "/tmp/supersu/su.zip");
+ui_print("Writing radio image...");
+package_extract_file("static_nvbk.bin", "/dev/block/platform/msm_sdcc.1/by-name/oppostanvbk");
+package_extract_file("tz.mbn", "/dev/block/platform/msm_sdcc.1/by-name/tz");
+package_extract_file("sbl1.mbn", "/dev/block/platform/msm_sdcc.1/by-name/sbl1");
+package_extract_file("sdi.mbn", "/dev/block/platform/msm_sdcc.1/by-name/sdi");
+package_extract_file("rpm.mbn", "/dev/block/platform/msm_sdcc.1/by-name/rpm");
+package_extract_file("emmc_appsboot.mbn", "/dev/block/platform/msm_sdcc.1/by-name/aboot");
+package_extract_file("logo.bin", "/dev/block/platform/msm_sdcc.1/by-name/LOGO");
+package_extract_file("NON-HLOS.bin", "/dev/block/platform/msm_sdcc.1/by-name/modem"); 
 unmount("/system");
